StabilityClient.Net/StabilityClient.Net/Extensions/AsyncServerStreamingCallExtensions.cs

```diff
@@ -63,4 +80,25 @@ public static class AsyncServerStreamingCallExtensions {
     private static GenerateResponseSaveResult CreateResponse(string path) => new() {
         FullPath = path
     };
-}
+
+    private static string GetFileName(string? fileName, string directoryPath) {
+        if (fileName is null) {
+            return $"{DateTime.Now:MM-dd-yy_HH-mm-ss}-{Guid.NewGuid()}.png";
+        }
+
+        var fullPath = Path.Combine(Path.GetFullPath(directoryPath), $"{fileName}.png");
+        if (File.Exists(fullPath)) {
+            return $"{fileName}-{Guid.NewGuid()}.png";
+        }
+
+        return $"{fileName}.png";
+    }
+
+    private static void ValidatePath(string path) {
+        if (string.IsNullOrEmpty(path)) {
+            throw new ArgumentException(
+                $"Directory path cannot be null or empty, was: {path}. Change value of {nameof(path)}.",
+                nameof(path));
+        }
+    }
+}
```

Comment:
Please add endline. ;) Maybe your IDE has an option to add them automatically for new files?

Reply:
I added endline and settings for it for future :)